gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -1396,10 +1396,46 @@ private void registerRoute(Buffer buffer) {
         // Configure connection pool size
         hook.setConnectionPoolSize(jsonHook.getInteger(HttpHook.CONNECTION_POOL_SIZE_PROPERTY_NAME));
 
-        routeRepository.addRoute(routedUrl, createRoute(routedUrl, hook));
+        boolean mustCreateNewRoute = true;
+        Route existingRoute = routeRepository.getRoutes().get(routedUrl);
+        if (existingRoute != null) {
```

Comment:
You actually could put the `null` - check in the `mustCreateNewRouteForHook` method. In case no hook is found, a new hook has to be created, therefor it would fit, don’t you think? Maybe you would have to rename the method, e.g. `mustCreateNewRoute`. But that's just a suggestion.

Reply:
Right - we could do it this way. If it's not a *must have* for you I would prefer to avoid this effort now.